src/manifest/mod.rs

```diff
@@ -241,6 +253,59 @@ impl CrateData {
         }
     }
 
+    /// Read the `manifest_path` file and deserializes it using the toml Deserializer.
+    /// Returns a Result containing `ManifestAndUnsedKeys` which contains `CargoManifest`
+    /// and a `BTreeSet<String>` containing the unused keys from the parsed file.
+    ///
+    /// # Errors
+    /// Will return Err if the file (manifest_path) couldn't be read or
+    /// if deserialize to `CargoManifest` fails.
+    pub fn parse_crate_data(manifest_path: &Path) -> Result<ManifestAndUnsedKeys, Error> {
+        let manifest = fs::read_to_string(&manifest_path)
+            .with_context(|_| format!("failed to read: {}", manifest_path.display()))?;
+        let manifest = &mut toml::Deserializer::new(&manifest);
+
+        let mut unused_keys = BTreeSet::new();
+        let mut has_possible_typo = false;
+        let levenshtein_threshold = 1;
+
+        let manifest: CargoManifest = serde_ignored::deserialize(manifest, |path| {
+            let path_string = path.to_string();
+
+            if levenshtein(WASM_PACK_METADATA_KEY, &path_string) == levenshtein_threshold {
```

Comment:
Could this perhaps have some more guards like testing that the key starts with `package.metadata` and then just running levenshtein over the key afterwards, ideally only matching "wasm-pack" up against the next part of the key?

Also, I think this may perhaps want to be `<= threshold` to ensure that exact matches are warned about

Reply:
Sounds good! Ah, ofc it should be `<=`.